src/main/java/org/apache/sling/feature/cpconverter/handlers/slinginitialcontent/BundleSlingInitialContentJarEntryExtractor.java

```diff
@@ -117,17 +115,21 @@ void extractAndAddToAssembler(@NotNull BundleSlingInitialContentExtractContext c
 
             }
 
-            try (Archive virtualArchive = SingleFileArchive.fromPathOrInputStream(tmpDocViewInputFile, bundleFileInputStream,
-                    () -> Files.createTempFile(context.getConverter().getTempDirectory().toPath(), "initial-content", Text.getName(file.getName())), contentPackageEntryPath)) {
-                // in which content package should this end up?
-
-                if (tmpDocViewInputFile != null) {
-                    packageAssembler.addEntry(contentPackageEntryPath, tmpDocViewInputFile.toFile());
-                } else {
-                    packageAssembler.addEntry(contentPackageEntryPath, bundleFileInputStream);
+            //if we are writing in a .content.xml file as default, 
+            // but the .content.xml is already present (written by specifications, then we don't do anything.)
+            if(!(isDefaultContentXmlFile && packageAssembler.alreadyPresent(contentPackageEntryPath))){
```

Comment:
nitpicking: you have a space missing after the if.

the serious part:
i am trying to map your comment above to this if-clause.... you only want to skip this step if `isDefaultContentXmlFile` is `true` and the entry is already present, right?
if `isDefaultContentXmlFile` is false you don't care about the collision and if the entry does not exist you can safely ignore the `isDefaultContentXmlFile`... is that what you are saying?

why? and how did it work before? because the check for alreadyPresent was not present before, right?
please elaborate in the comment for the next dev looking at this being able to follow you